syntax = "proto3";

package employee;

// Interface exported by the server.
service Employee {
    rpc Read(readRequestPB) returns (readResponseListPB) {}
}

message readRequestPB {
    string first_name = 1;
    string last_name = 2;
    string email = 3;
    string gender = 4;
    string ip_address = 5;
    string country = 6;
    string postcode = 7;
}

message readResponsePB {
    string first_name = 1;
    string last_name = 2;
    string email = 3;
    string gender = 4;
    string ip_address = 5;
    string country = 6;
    string postcode = 7;
    int32 id = 8;
}

message readResponseListPB {
    repeated readResponsePB employee = 1;
}